debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlHistoryRecordComparator.java

```diff
@@ -145,4 +143,79 @@ else if (recordedGtidSetStr != null) {
         // The binlog coordinates are the same ...
         return true;
     }
+
+    public static class BinlogFilename implements Comparable<BinlogFilename> {
+        final private String baseFilename;
+        final private long extension;
+
+        private BinlogFilename(String baseFilename, long extension) {
+            this.baseFilename = baseFilename;
+            this.extension = extension;
+        }
+
+        public static BinlogFilename of(String filename) {
+            if (!filename.contains(".")) {
+                throw new IllegalArgumentException("Invalid binlog filename format: " + filename);
+            }
+
+            int index = filename.lastIndexOf(".");
+            String baseFilename = filename.substring(0, index);
+            String stringExtension = filename.substring(index + 1);
+
+            try {
+                long extension = Long.parseLong(stringExtension);
+                return new BinlogFilename(baseFilename, extension);
+            }
+            catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Can't parse binlog filename extension: " + filename);
+            }
+        }
+
+        public String getBaseFilename() {
```

Comment:
This and the following getter aren't necessary. Objects have access to the private properties of other objects of the same class.